src/manifest.rs

```diff
@@ -44,6 +48,16 @@ pub enum OciManifest {
     ImageIndex(OciImageIndex),
 }
 
+impl OciManifest {
+    /// Returns the appropriate content-type for each variant.
+    pub fn content_type(&self) -> &str {
+        match self {
+            OciManifest::Image(_) => OCI_IMAGE_MEDIA_TYPE,
```

Comment:
Nit: would it be better to return this as an actual `HeaderValue` rather than having to do the `parse().unwrap()` outside of this function?

Reply:
I opted to keep it easily comparable to the content-type constants, but it was a coin-toss decision.

It would be nice to have an enum for the content-types and represent this as such, but that's a different topic.